presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,42 +179,79 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        int bestChoiceIndex = 0;
+        boolean[] isInvalid = new boolean[choices.size()];
+        for (int i = 0; i < choices.size(); i++) {
+            ScalarImplementationChoice current = choices.get(i);
+            for (int j = 0; j < arguments.size(); j++) {
+                boolean isInputReference = false;
+                if (arguments.get(j) instanceof InputReferenceNode) {
+                    isInputReference = true;
+                }
+                NullConvention currentNullConvention = current.getArgumentProperty(j).getNullConvention();
+                if (isInputReference && currentNullConvention == NullConvention.BLOCK_AND_POSITION && !isInvalid[i]) {
+                    bestChoiceIndex = i;
```

Comment:
The logic of this nested loop is incorrect. As written, this code could update `bestChoiceIndex` after you have only looked at the first argument. The code couldn't possibly know that a choice is eligible before looking at all the argument, right?